backend/src/lib.rs

```diff
@@ -136,6 +180,22 @@ pub type Proof = Vec<u8>;
 
 /// Dynamic interface for a backend.
 pub trait Backend<F: FieldElement> {
+    fn add_verification_key(
+        &mut self,
+        pil: &Analyzed<F>,
+        fixed: &[(String, Vec<F>)],
+        vkey: Vec<u8>,
+    );
+
```

Comment:
Ideally I wanted to add this function as `with_verification_key(self, ...) -> Self` but that won't compile because of the dynamic dispatcher.

Reply:
Did you mean `with_verification_key(...) -> Self` ? There is a factory class to create backends, and it has `create(&self, degree: DegreeType)` and `create_from_setup(&self, input: &mut dyn io::Read)`.

We could add `fn create_with_verification_key(...)` and `fn create_with_setup_and_verification_key(...)`, but that is too combinatoric for my taste.

I propose instead just having one `fn create(...)`, taking as `Option<...>` the setup and the verification key.

Reply:
> but that is too combinatoric for my taste.

yes that's also what I thought

> Did you mean with_verification_key(...) -> Self

No I meant an object that's already created (with `create` or `creat_from_setup` and you can just append the verification key

Reply:
In the end, do you want the object to be created with the verification key or not? Why not just pass it with `verify()`?

Reply:
I don't have a strong opinion about it. I'd be fine without it being created with it. The reason not to pass it with `verify` is that we also need `params` in verify. So I found it easier to already have everything in the object.